lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1175,73 +1175,60 @@ bool SessionState::reset_reporting_charging_credit(
+  auto credit_uc = get_credit_uc(CreditKey(update), session_uc);
   if (!update.success()) {
     // update unsuccessful, reset credit and return
-    MLOG(MDEBUG) << session_id_ << " Received an unsuccessful update for RG "
-                 << update.charging_key();
+    MLOG(MERROR) << "Credit update failed RG:" << key << " for " << session_id_;
     grant->credit.mark_failure(update.result_code(), credit_uc);
     if (grant->should_deactivate_service()) {
       grant->set_service_state(SERVICE_NEEDS_DEACTIVATION, *credit_uc);
     }
     return false;
   }
-  MLOG(MINFO) << session_id_ << " Received a charging credit for RG: "
-              << update.charging_key();
+  MLOG(MINFO) << "Received a credit RG:" << key << " for " << session_id_;
   grant->receive_charging_grant(update.credit(), credit_uc);
 
   if (grant->reauth_state == REAUTH_PROCESSING) {
     grant->set_reauth_state(REAUTH_NOT_NEEDED, *credit_uc);
   }
   if (!grant->credit.is_quota_exhausted(1) &&
       grant->service_state != SERVICE_ENABLED) {
-    // if quota no longer exhausted, reenable services as needed
-    MLOG(MINFO) << "Quota available. Activating service";
+    // if quota no longer exhausted, re-enable services as needed
+    MLOG(MINFO) << "New quota now available. Activating service RG:" << key
+                << " for " << session_id_;
     grant->set_service_state(SERVICE_NEEDS_ACTIVATION, *credit_uc);
   }
-  return contains_credit(update.credit().granted_units()) ||
-         is_infinite_credit(update);
+  return true;
 }
 
 bool SessionState::init_charging_credit(
     const CreditUpdateResponse& update,
-    SessionStateUpdateCriteria& update_criteria) {
+    SessionStateUpdateCriteria& session_uc) {
+  const uint32_t key = update.charging_key();
   if (!update.success()) {
     // init failed, don't track key
-    MLOG(MERROR) << "Credit init failed for imsi " << imsi_
-                 << " and charging key " << update.charging_key();
+    MLOG(MERROR) << "Credit init failed RG:" << key << " for " << session_id_;
     return false;
   }
-  MLOG(MINFO) << session_id_ << " Initialized a charging credit for RG: "
-              << update.charging_key();
-
-  auto charging_grant    = std::make_unique<ChargingGrant>();
-  charging_grant->credit = SessionCredit(SERVICE_ENABLED, update.limit_type());
-
-  charging_grant->receive_charging_grant(update.credit());
-  update_criteria.charging_credit_to_install[CreditKey(update)] =
-      charging_grant->marshal();
-  credit_map_[CreditKey(update)] = std::move(charging_grant);
-  return contains_credit(update.credit().granted_units()) ||
-         is_infinite_credit(update);
-}
-
-bool SessionState::contains_credit(const GrantedUnits& gsu) {
-  return (gsu.total().is_valid() && gsu.total().volume() > 0) ||
-         (gsu.tx().is_valid() && gsu.tx().volume() > 0) ||
-         (gsu.rx().is_valid() && gsu.rx().volume() > 0);
-}
-
-bool SessionState::is_infinite_credit(const CreditUpdateResponse& response) {
```

Comment:
aren't we checking this anymore

Reply:
@uri200 The `is_inifinte_credit` function only existed because we were rejecting credits with empty GSU. So even if a credit was empty, if the credit was inifite, we would want to install it. But now that we are installing all successful credit, we don't need this check.